LandSandBoat/server
# Move addEffect lvCorrect/Stat handling to modifier based for flexibility

**_I affirm:_**

## What does this pull request do?
Move addEffect lvCorrect/Stat handling to modifier based for flexibility...Instead of assuming all ranged attacks will be the same.

Additionally:
- Moved up the lv correction call to before we check for proc
- changed a small number of the sql comments to use the enum for readability. More to come later.
- add some additional notation commenting

## Steps to test these changes
1. Set your self immortal, give yourself a cross bow and some bloody bolts, and lower your hp so you can see the drains proc.
2. Pick out some monster and set your own level accordingly to test both being above and below the monsters level
3. GM yourself some INT gear to swap in and out to see its changing the damage/drain amount, likewise can debuff the mob (shock, absorb INT) to change the difference.
4. add some prints to see proc rate changing
5. Blast that monster right between the eyes, see drain numbers
6. Forgive Teo for that placeholder modifier sin. Teo hates placeholdering, and promises it will get used, but will remove it on request anyway
